Update docs for DSL2 release (#818)
nf-core/nf-co.re: markdown/developers/adding_pipelines.md

* `docs/`: Markdown files for documenting the pipeline

> * `lib/`: The lib directory contains groovy utility functions. These are called from within the nf-core pipeline to do common pipeline tasks, like formatting the header for the command line. Any larger groovy function you want to call in the code should therefore go in here. Currently, the following files are included:

Reviewer: ```suggestion
* `lib/`: The lib directory contains Groovy utility functions. These are called from within the nf-core pipeline to do common pipeline tasks (e.g. parameter schema validation) and to hold Groovy functions that may be useful in the pipeline context (e.g. to validate pipeline-specific parameters). Currently, the following files are included:
```

>   * `NfcoreSchema.groovy`
>   * `NfcoreTemplate.groovy`
>   * `Utils.groovy`
>   * `WorkflowMain.groovy`
>   * `WorkflowPipeline.groovy`
>   * `nfcore_external_java_deps.jar`

Reviewer 2: ```suggestion
  * `NfcoreSchema.groovy` - Functions to validate input parameters using the pipeline JSON schema
  * `NfcoreTemplate.groovy` - Additional nf-core specific pipeline functions (sending emails, checking nf-core config profiles and more)
  * `Utils.groovy` - Additional generic pipeline functions (checking conda config and more)
  * `WorkflowMain.groovy` - Startup functions for the main pipeline (printing logs, custom params initialisation and more)
  * `WorkflowPipeline.groovy` - Functions for pipeline subworkflows
  * `nfcore_external_java_deps.jar` - Bundled Groovy dependencies so that pipelines work offline (mostly for JSON schema validation - see imports in `NfcoreSchema.groovy`)
```

* `modules/`: Contains pipeline-specific and common nf-core modules

* `workflows/`: Contains the main pipeline workflows to be executed in the `main.nf` file

> * `subworkflows/`: Contains smaller subworkflows that typically consist out of a few modules chained together

Reviewer: ```suggestion
* `subworkflows/`: Contains smaller subworkflows that typically consist of a few modules chained together
```
